source/firmware/arch/timing-manager.md

```diff
@@ -0,0 +1,170 @@
+# Timing Manager
+
+This page serves to document the operation and configuration of the AMDC Timing Manager, a system-level IP responsible for aligning sensor sampling, feedback, and control task execution to the PWM-carrier peak/valley for predicatable control and EMI minimization.
+
+The primary audience of this documentation is the AMDC development/maintainers team, but can also be a good reference for users of the platforms seeking to better understand the timing system of the AMDC to better optimize their control.
+
+## Behaviour
+
+The Timing Manager has the followings logical inputs and outputs:
+
+| **Inputs**                    | **Outputs**         |
+|-------------------------------|---------------------|
+| PWM Carrier (peaks & valleys) | Sensor Trigger      |
+| User Event Ratio              | Scheduler Interrupt |
+| Sensor Enable Statuses        |                     |
+| Sensor Done Statuses          |                     |
+
+### Inputs
+
+#### PWM Carrier
+
+The timing manager receives the peaks and valleys of the switching PWM triangle carrier, as these times are when EMI noise is least-disruptive to the sensor interfaces. The user can configure the timing manager to trigger sensor sampling on every _N_ peaks and/or valleys. The default is 100 kHz PWM frequency, with the timing manager sending a trigger every 10 valleys, or at a 10 kHz frequency.
+
+#### User Event Ratio
+
+The User Event Ratio is the ratio "_every N_" PWM carrier events should trigger the sensors and control task; i.e. the user could override the default ratio of 10, and request that the sensors instead be sampled every 20 valleys. 
+
+#### Sensor Enable Statuses
+
+The timing manager must wait to schedule control tasks until all the user's desired sensors have reported back their data. Therefore the user must enable all their desired sensors within the timing manager so it knows which sensors to wait for and which to ignore.
+
+#### Sensor Done Statuses
+
+Each enabled sensor will be triggered to sample by the timing manager at qualifying PWM carrier events. When a sensor is triggered, its `done` status is reset to 0, until that sensor completes its sampling, makes its new data available, and re-asserts its `done` status to 1 for the timing manager. 
+
+```{important}
+When creating new drivers for sensors, the sensor driver's done signal that is reported back to the Timing Manager MUST:
+
+- Be reset to 1
+- AND guarantee that the will be done eventually (possibly even time-out)
```

Comment:
I find the phrasing on these bullets a bit confusing. Part of it is that above this, we have the phrase "reset to 0" vs "re-asserts... to 1"

I understand that we are trying to tell the developer of new drivers that the must eventually indicate a done status or the timing manager will get stuck. 

Can we think of alternate text for these bullets?